expr/func_summarize_test.go

```diff
@@ -665,6 +665,78 @@ func TestSummarizeAlignToFrom(t *testing.T) {
 	testSummarize("AlignToFrom", input, outputMax[1], "45s", "max", true, t)
 }
 
+func TestSummarizeLargeIntervalTimestamps(t *testing.T) {
+
+	// This test is specifically testing the timestamp behavior, so we don't need real values.
+	// However, we do need a lot of datapoints to trigger the bug we are regression testing against.
+	interval1Day := uint32(24 * 60 * 60)
+	numTimestamps := uint32(2000)
+	startTime := uint32(1464637518)
+	endTime := startTime + numTimestamps*interval1Day
+
+	inputDps := make([]schema.Point, 0, numTimestamps)
+
+	for i := uint32(0); i < numTimestamps; i++ {
+		inputDps = append(inputDps, schema.Point{Val: 0, Ts: startTime + i*interval1Day})
+	}
+
+	outputInterval := 365 * interval1Day
+
+	var unaligned365dsum = []schema.Point{
+		{Val: 0, Ts: 1450656000},
+		{Val: 0, Ts: 1482192000},
+		{Val: 0, Ts: 1513728000},
+		{Val: 0, Ts: 1545264000},
+		{Val: 0, Ts: 1576800000},
+		{Val: 0, Ts: 1608336000},
+	}
+
+	var aligned365dsum = []schema.Point{
+		{Val: 0, Ts: 1464637518},
+		{Val: 0, Ts: 1496173518},
+		{Val: 0, Ts: 1527709518},
+		{Val: 0, Ts: 1559245518},
+		{Val: 0, Ts: 1590781518},
+		{Val: 0, Ts: 1622317518},
+	}
+
+	input := []models.Series{
+		{
+			Target:     "align",
+			QueryPatt:  "align",
+			QueryFrom:  startTime,
+			QueryTo:    endTime,
+			Interval:   outputInterval,
+			Datapoints: getCopy(inputDps),
+		},
+	}
+	outputSum := [][]models.Series{
+		{
+			{
+				Target:     "summarize(align, \"365d\", \"sum\")",
+				QueryPatt:  "summarize(align, \"365d\", \"sum\")",
```

Comment:
any particular reason we're testing with 365d buckets?
i would suggest to target the bugreport specifically and use 30d buckets , and a from-until range as used in the ticket of 89.000683 days, as well as a few different values like 89 days, 90 days, etc

Reply:
The bug is also dependent on the input datapoints. That wasn't specified in the bug report, but I don't believe it's necessary to exactly reproduce as in the bug report. I expect that I could though, if you wish.

Reply:
FWIW the data from the bug report has interval=10s